### PR TITLE
Support Kotlin 1.9.20

### PR DESCRIPTION
**Pull request by Daeda88**

### CODE REVIEW

**Review by dasdranagon — APPROVED** [review]

**Review by thoutbeckers — APPROVED** [review]

File: firebase-database/src/androidMain/kotlin/dev/gitlive/firebase/database/database.kt [hunk 1]
```diff
@@ -38,7 +38,7 @@ suspend fun <T> Task<T>.awaitWhileOnline(database: FirebaseDatabase): T =
             .reference(".info/connected")
             .valueEvents
             .debounce(2.seconds)
-            .filter { !it.value<Boolean>() }
+            .filterNot { it.value<Boolean>() }
```
Comment (avdyushin) [inline firebase-database/src/androidMain/kotlin/dev/gitlive/firebase/database/database.kt:41]: 🥂

File: firebase-auth/src/commonTest/kotlin/dev/gitlive/firebase/auth/auth.kt [hunk 2]
```diff
@@ -127,9 +134,9 @@ class FirebaseAuthTest {
             user!!.uid
         }
 
-        check(Firebase.auth.currentUser != null)
-        Firebase.auth.signOut()
-        check(Firebase.auth.currentUser == null)
+        check(auth.currentUser != null)
```
Comment (avdyushin) [inline firebase-auth/src/commonTest/kotlin/dev/gitlive/firebase/auth/auth.kt:137]: BTW, what is `check` compare to `assertNull`?

Comment (avdyushin) [reply firebase-auth/src/commonTest/kotlin/dev/gitlive/firebase/auth/auth.kt:137]: Here is also `checkNotNull` available in anycase :)

File: firebase-database/src/androidMain/kotlin/dev/gitlive/firebase/database/database.kt (hunk 1 above)
Comment (Daeda88) [reply firebase-database/src/androidMain/kotlin/dev/gitlive/firebase/database/database.kt:41]: To be fair, this is from the upstream :P

**Pull request merged**